examples/amm/README.md

```diff
@@ -0,0 +1,147 @@
+<!-- cargo-rdme start -->
+
+# Automated Market Maker (AMM) Example Application
+
+This example implements an Automated Market Maker (AMM) which demonstrates DeFi capabilities of the 
+Linera protocol. Pre-requisites for the AMM application is the `funglible` application, as we will
+be adding, removing liquidity and also perform a swap.
+
+# How it works 
+
+It supports the following operations
+
+- Swap: For a given input token and an input amount, it swaps that token amount for an 
+amount of the other token calculated based on the current AMM ratio. Note: The `Swap` operations 
+are needed to performed from a remote chain 
+
+- Add Liquidity: This operation allows adding liquidity to the AMM. Given a maximum 
+`token0` and `token1` amount that you're willing to add, it adds liquidity such that you'll be 
+adding at most `max_token0_amount` of `token0` and `max_token1_amount` of `token1`. The amounts 
+will be calculated based on the current AMM ratio. The owner, in this case, refers to the user 
+adding liquidity, which currently can only be a chain owner.
+
+- Remove Liquidity: This is a liquidity removal operation. Given a token idx of the token you'd 
+like to remove (can be 0 or 1), and an amount of that token that you'd like to remove, it calculates 
+how much of the other token will also be removed based on the current AMM ratio. Then it removes 
+the amounts from both tokens as a removal of liquidity. The owner, in this context, is the user 
+removing liquidity, which currently can only be a chain owner.
+
+# Usage
+
+## Setting Up
+
+First, build Linera and add it to the path:
+
+```bash
+cargo build
+export PATH=$PWD/target/debug:$PATH
+```
+
+To start the local linera network
+
+```bash
+linera net up --testing-prng-seed 37
+```
+
+This will start the local linera network. We used the
+test-only CLI option `--testing-prng-seed` to make keys deterministic and simplify our
+presentation.
+
+Copy and paste `export LINERA_WALLET="/var . . . ."` & `export LINERA_STORAGE="rocksdb:/. . . ."` from output of `linera net up --testing-prng-seed 37` into another terminal
+
+```bash
+OWNER_1=e814a7bdae091daf4a110ef5340396998e538c47c6e7d101027a225523985316
+OWNER_2=453690095cdfe6dbde7fc577e56bb838a7ee7920a72512d4a87748b4e151ed61
+CHAIN_1=e476187f6ddfeb9d588c7b45d3df334d5501d6499b3f9ad5595cae86cce16a65
+CHAIN_2=e54bdb17d41d5dbe16418f96b70e44546ccd63e6f3733ae3c192043548998ff3
+```
+
+Now we have to publish and deploy the AMM application,
+
+```bash
+appid1=YOUR_FUNGIBLE_APP1_ID
+appid2=YOUR_FUNGIBLE_APP2_ID
```

Comment:
Aren't the app IDs deterministic, too? I'd include the commands to deploy the fungible apps in this README.

(Is there a reason you use lowercase for these?)

Reply:
Something like this?

```
FUN1_APP_ID=$(linera publish-and-create ../target/wasm32-unknown-unknown/release/fungible_{contract,service}.wasm \
    --json-argument "{ \"accounts\": {
        \"User:$OWNER_1\": \"100.\",
        \"User:$OWNER_2\": \"150.\"
    } }" \
    --json-parameters "{ \"ticker_symbol\": \"FUN1\" }" \
)
```

Reply:
Yes, that looks right. :+1: